src/Avalonia.Controls/Control.cs

```diff
@@ -372,23 +372,19 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
         /// </summary>
         public virtual void InvalidateMirrorTransform()
         {
-            var flowDirection = this.FlowDirection;
+            var flowDirection = FlowDirection;
             var parentFlowDirection = FlowDirection.LeftToRight;
 
             bool bypassFlowDirectionPolicies = BypassFlowDirectionPolicies;
             bool parentBypassFlowDirectionPolicies = false;
 
-            var parent = this.FindAncestorOfType<Control>();
+            var parent = ((IVisual)this).VisualParent as Control;
+
             if (parent != null)
             {
                 parentFlowDirection = parent.FlowDirection;
                 parentBypassFlowDirectionPolicies = parent.BypassFlowDirectionPolicies;
             }
-            else if (Parent is Control logicalParent)
```

Comment:
I do not know what you are planning, but it is consumed in case the element is not connected to the visual tree like `ComboBox` etc.

Reply:
Value inheritance always works in that case

Reply:
but the calculation will be wrong because the element is not connected to the visual tree. ComboBoxItem never no connected to visual tree (in closed mode), how did he know whether to apply MirrorTransform if he did not have a parent?

Reply:
The moment the item is visible it is connected and therefore has a parent visual. There is no need to compute anything when the control isn't visible.

Reply:
I don't see any public use for HasMirrorTransform

Reply:
ComboBoxItem hasn't parent, ComboBox render him with VisualBrush.

Reply:
just enter to ComboBox page with your PR and you will see.

Reply:
I don't see a difference to current master in combination with ComboBox. The arrow isn't rendered with current master as well.

Reply:
> the "hidden arrow within ComboBox" issue is fixed by #8174

i meant to content of ComboBox.

Reply:
Give me a few hours, I will add a few unit tests so we can avoid mistakes and to clarify the points.

Reply:
I'm missing something in ComboBox, trying to figure it out...

Reply:
Open up a new PR and I will review it shortly